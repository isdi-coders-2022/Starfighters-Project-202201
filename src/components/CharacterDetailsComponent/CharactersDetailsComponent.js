import CharacterDetailsBox from "./CharacterDetailsBox";

const CharactersDetailsComponent = ({ fighter }) => {
  return (
    <CharacterDetailsBox className="character-details">
      <img
        className="character-details__image"
        alt="character profile"
        src={fighter.image}
        width="400"
      />
      <div className="light-div"></div>
      <div className="section-div">
        <h2 className="character-details__name">{fighter.name}</h2>
        <section className="character-details__info">
          <p>Height:{fighter.height}</p>
          <p>Mass: {fighter.mass}</p>
          <p>Homeworld: {fighter.homeworld}</p>
          <p>Species: {fighter.species}</p>
          <p>
            <a href={fighter.wiki}>More info:</a>
          </p>
        </section>
      </div>
    </CharacterDetailsBox>
  );
};

export default CharactersDetailsComponent;
