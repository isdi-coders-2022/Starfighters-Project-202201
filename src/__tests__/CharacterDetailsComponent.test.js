import { render, screen } from "@testing-library/react";
import reactTestRenderer from "react-test-renderer";
import CharactersDetailsComponent from "../components/CharacterDetailsComponent/CharactersDetailsComponent";

describe("Given a CharacterDetailsComponent component", () => {
  describe("When it's rendered", () => {
    test("Then it should display an <h>", () => {
      const fighter = {
        id: 1,
        name: "Luke Skywalker",
        height: 1.72,
        mass: 73,
        gender: "male",
        homeworld: "tatooine",
        wiki: "http://starwars.wikia.com/wiki/Luke_Skywalker",
        image:
          "https://vignette.wikia.nocookie.net/starwars/images/2/20/LukeTLJ.jpg",
        born: -19,
        bornLocation: "polis massa",
        died: 34,
        diedLocation: "ahch-to",
        species: "human",
        hairColor: "blond",
        eyeColor: "blue",
        skinColor: "light",
        cybernetics: "Prosthetic right hand",
        affiliations: [
          "Alliance to Restore the Republic",
          "Red Squadron",
          "Rogue Squadron",
          "Massassi Group",
          "Leia Organa's team",
          "Endor strike team",
          "Jedi Order",
          "Bright Tree tribe",
          "New Republic",
          "Resistance",
        ],
        masters: ["Obi-Wan Kenobi", "Yoda"],
        apprentices: [
          "Leia Organa",
          "Ben Solo (along with a dozen apprentices)",
          "Rey",
        ],
        formerAffiliations: [],
      };

      render(<CharactersDetailsComponent fighter={fighter} />);

      const expectedOutput = screen.queryByRole("heading");

      expect(expectedOutput).toBeInTheDocument();
    });
  });

  describe("When executing the component", () => {
    test("Then it should render", () => {
      const fighter = {
        id: 1,
        name: "Luke Skywalker",
        height: 1.72,
        mass: 73,
        gender: "male",
        homeworld: "tatooine",
        wiki: "http://starwars.wikia.com/wiki/Luke_Skywalker",
        image:
          "https://vignette.wikia.nocookie.net/starwars/images/2/20/LukeTLJ.jpg",
        born: -19,
        bornLocation: "polis massa",
        died: 34,
        diedLocation: "ahch-to",
        species: "human",
        hairColor: "blond",
        eyeColor: "blue",
        skinColor: "light",
        cybernetics: "Prosthetic right hand",
        affiliations: [
          "Alliance to Restore the Republic",
          "Red Squadron",
          "Rogue Squadron",
          "Massassi Group",
          "Leia Organa's team",
          "Endor strike team",
          "Jedi Order",
          "Bright Tree tribe",
          "New Republic",
          "Resistance",
        ],
        masters: ["Obi-Wan Kenobi", "Yoda"],
        apprentices: [
          "Leia Organa",
          "Ben Solo (along with a dozen apprentices)",
          "Rey",
        ],
        formerAffiliations: [],
      };
      const component = reactTestRenderer.create(
        <CharactersDetailsComponent fighter={fighter} />
      );
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
