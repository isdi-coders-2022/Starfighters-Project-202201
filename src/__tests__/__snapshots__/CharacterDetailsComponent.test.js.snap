// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a CharacterDetailsComponent component When executing the component Then it should render 1`] = `
<div
  className="sc-bdvvtL hbhmpE character-details"
>
  <img
    alt="character profile"
    className="character-details__image"
    src="https://vignette.wikia.nocookie.net/starwars/images/2/20/LukeTLJ.jpg"
    width="400"
  />
  <div
    className="light-div"
  />
  <div
    className="section-div"
  >
    <h2
      className="character-details__name"
    >
      Luke Skywalker
    </h2>
    <section
      className="character-details__info"
    >
      <p>
        Height:
        1.72
      </p>
      <p>
        Mass: 
        73
      </p>
      <p>
        Homeworld: 
        tatooine
      </p>
      <p>
        Species: 
        human
      </p>
      <p>
        <a
          href="http://starwars.wikia.com/wiki/Luke_Skywalker"
        >
          More info:
        </a>
      </p>
    </section>
  </div>
</div>
`;
